File: 2448-count-number-of-bad-pairs/count-number-of-bad-pairs.java
class Solution {
    public long countBadPairs(int[] nums) {
        int n=nums.length;
        
        long ans=0;

        HashMap<Integer,Integer> map = new HashMap<>();

        for(int i=0;i<n;i++){
            int key=nums[i]-i;
            
            ans+=(long)(i-map.getOrDefault(key,0));

            map.put(key,map.getOrDefault(key,0)+1);

        }

        return ans;


        
    }
}

File: 2448-count-number-of-bad-pairs/count-number-of-bad-pairs.cpp
class Solution {
public:
    long long countBadPairs(vector<int>& nums) {
        map<int,int>mp;

        long long ans=0;

        for(int i=0;i<nums.size();i++){
            long long toFd=nums[i]-i;

            ans+=(i-mp[toFd]);

            mp[toFd]++;
        }

        return ans;
    }
};

